registry/registry_control.go

```diff
@@ -0,0 +1,242 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package registry
+
+import (
+	"fmt"
+	"regexp"
+	"slices"
+	"strings"
+)
+
+var (
+	validAccountID = regexp.MustCompile("^(?:[a-z0-9A-Z]{32}|[-a-z0-9]{2,28})$")
+)
+
+type AuthenticationMethod int
+
+const (
+	OperatorKey AuthenticationMethod = iota
+	Store
+)
+
+var authMethodMap = map[string]AuthenticationMethod{
+	"operator-key": OperatorKey,
+	"store":        Store,
+}
+
+// StringToAuthenticationMethod converts a string to AuthenticationMethod.
+func StringToAuthenticationMethod(s string) (AuthenticationMethod, error) {
+	if method, ok := authMethodMap[s]; ok {
+		return method, nil
+	}
+	return -1, fmt.Errorf("unknown authentication method: %s", s)
+}
+
+var authMethodStringMap = map[AuthenticationMethod]string{
+	OperatorKey: "operator-key",
+	Store:       "store",
+}
+
+// String prints the string representation of AuthenticationMethod.
+func (auth AuthenticationMethod) String() string {
+	if s, ok := authMethodStringMap[auth]; ok {
+		return s
+	}
+	return "unknown"
+}
+
+// Operator holds the delegations for a single operator.
+type Operator struct {
+	OperatorID string
```

Comment:
nitpick: we don't need to repeat operator in this name
```suggestion
	ID string
```